### PR TITLE
Avoid unsupported attributes when not building with UBSAN

### PR DESCRIPTION
**Pull request by tamird**
@yiwu-arbug see individual commits.

### CODE REVIEW

**Comment by tamird** [comment]
@yiwu-arbug this is green, can you have a look?

File: db/db_test_util.h [hunk 1]
```diff
@@ -49,10 +49,7 @@
 #include "util/mutexlock.h"
 
 #include "util/string_util.h"
-// SyncPoint is not supported in Released Windows Mode.
```
Comment (yiwu-arbug) [inline db/db_test_util.h]: Did you test with release build with Windows?

Comment (tamird) [inline db/db_test_util.h]: What do you mean? The entire file is behind a guard:

https://github.com/facebook/rocksdb/blob/master/util/sync_point.h#L28
https://github.com/facebook/rocksdb/blob/master/util/sync_point.cc#L17

Comment (yiwu-arbug) [inline db/db_test_util.h]: Right.. I don't know why we have the comment saying it not being support in release mode..

File: util/coding.h [hunk 2]
```diff
@@ -354,6 +354,13 @@ inline Slice GetSliceUntil(Slice* slice, char delimiter) {
 }
 
 template<class T>
+#ifdef ROCKSDB_UBSAN_RUN
+#if defined(__clang__)
+__attribute__((no_sanitize("alignment")))
```
Comment (yiwu-arbug) [inline util/coding.h]: should it be `__no_sanitize__`? I don't have experience with it but out linter is complaining about it.

File: db/db_test_util.h (hunk 1 above)
Comment (tamird) [inline db/db_test_util.h]: Indeed. Looks like detritus, which is why I removed it :)

File: util/coding.h (hunk 2 above)
Comment (tamird) [inline util/coding.h]: I can change it - I don't really understand the difference, all the documentation I could find mentioned `no_sanitize`, not `__no_sanitize__` (https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#disabling-instrumentation-with-attribute-no-sanitize-undefined).

Anyway, changed.

**Comment by yiwu-arbug** [comment]
Merging. Thank you for your contribution!

**Pull request closed without merging**